server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -1628,4 +1628,15 @@ public void calculateSharedForOwnerPools(Owner owner) {
         q.setParameter("owner", owner);
         q.executeUpdate();
     }
+
+    @Transactional
+    public List<Pool> listSharedPoolsFromMasterPool(Pool pool) {
```

Comment:
Please avoid using the work "master pool" in this function name, as the pool that was shared could have been a non-master ( non-normal ) bonus pool as well.. 
my recommendation : listSharedPoolsOf(Pool pool) {